woodwork/tests/datatable/test_datatable_indexing.py

```diff
@@ -162,6 +162,34 @@ def test_underlying_index_on_update(sample_df):
     assert type(dt._dataframe.index) == pd.Int64Index
     assert type(dt.to_dataframe().index) == pd.Int64Index
 
+    actual = dt.iloc[[0, 1]]
+    assert type(actual._dataframe.index) == pd.Index
+    assert type(actual.to_dataframe().index) == pd.Index
```

Comment:
A quirk comes up when setting the underlying index after using iloc of the datatable. Pandas converts Int64 to objects but preserves the dtype if set as int64.

```
   id         full_name
0   2       James Brown
1   3  Ms. Paige Turner
```
```python
assert str(df.id.dtype) == 'int64'
df.set_index('id', inplace=True)
assert str(df.index.dtype) == 'int64'
```
```python
df.reset_index(inplace=True)
df['id'] = df['id'].astype('Int64')
df.set_index('id', inplace=True)
assert str(df.index.dtype) == 'object'
```